### PR TITLE
Chart: Use stable API versions where available

### LINKED ISSUE #17188: Chart: Uses deprecated v1beta1 Ingress Resource. Incompatible with Kubernetes 1.22
**Apache Airflow version**: 2.1.1

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): 1.20

**Environment**: Amazon EKS deployed via Helm Chart v1.0.0

**What happened**: The ingress resource created via the Helm chart uses the now deprecated v1beta Ingress which will be removed in Kubernetes 1.22. The stable version has been available since 1.19 which is two versions behind, and at least one version behind on most major cloud providers.

**What you expected to happen**: The stable v1 resource with option for ingressClassName is used instead. If you wish to support Kubernetes versions older than 1.19 then Helm is capable of doing different things depending on detected Kubernetes versions, which I've seen other charts do.

### PR DESCRIPTION
**Pull request by cablespaghetti**
Fixes #17188 

Uses stable versions of Ingress, PodDisruptionBudget and CronJob APIs where available. There is no change for users of Kubernetes versions older than 1.19. Also tidies up some inconsistencies between the two Ingresses and removes variables on the flower Ingress which didn't do anything.

### CODE REVIEW

**Comment by potiuk** [comment]
CC: @jedcunningham @kaxil WDYT ?

**Review by mik-laj — APPROVED** [review]

**Review by jedcunningham — CHANGES_REQUESTED** [review]
Overall LGTM, thanks @cablespaghetti! Can you add test coverage in `chart/tests` as well though?

**Comment by cablespaghetti** [comment]
> Overall LGTM, thanks @cablespaghetti! Can you add test coverage in `chart/tests` as well though?

Sorry for the delay in this. I will get to this as soon as I can. I was having some trouble getting a working dev environment on my M1 Mac initially but I'm sure it won't be that hard to resolve.

**Comment by renan** [comment]
New versions of Ingress controllers (eg: `ingress-nginx`) are dropping support for `v1beta1` and Airflow is blocking us from updating to them.

Is there an ETA for this PR?

**Comment by kaxil** [comment]
> New versions of Ingress controllers (eg: `ingress-nginx`) are dropping support for `v1beta1` and Airflow is blocking us from updating to them.
> 
> Is there an ETA for this PR?

As soon as @cablespaghetti adds the tests :)

**Comment by cablespaghetti** [comment]
Really sorry this has taken so long, I have been super busy both at work out and outside. I will put in some time to try and get this done tomorrow.

**Comment by cablespaghetti** [comment]
Ok hopefully I've done enough now. Also upgraded the default it tests against to 1.22. Rerunning everything locally to check this hasn't broken anything at the moment.

File: chart/values.yaml [hunk 1]
```diff
@@ -120,24 +129,24 @@ ingress:
     annotations: {}
 
     # The path for the flower Ingress
-    path: ""
+    path: "/"
```
Comment (cablespaghetti) [inline chart/values.yaml:132]: Both path and pathType are mandatory in the stable api. I don't think setting it by default is likely break many people's config but thought I'd point it out.

**Comment by cablespaghetti** [comment]
I appreciate I can't complain as I took a very long time to complete the work on this PR. However just a reminder to @kaxil and @jedcunningham that this should be ready to go now. 👍

File: chart/values.schema.json [hunk 2]
```diff
@@ -195,16 +221,6 @@
...
-                        },
-                        "precedingPaths": {
-                            "description": "HTTP paths to add to the flower Ingress before the default path.",
-                            "type": "array",
-                            "default": []
-                        },
-                        "succeedingPaths": {
-                            "description": "HTTP paths to add to the flower Ingress after the default path.",
-                            "type": "array",
-                            "default": []
```
Comment (jedcunningham) [inline chart/values.schema.json:207]: Can you add a note to `chart/UPDATING.rst` about these removals?

Comment (cablespaghetti) [reply chart/values.schema.json:207]: Done 👍

**Comment by cablespaghetti** [comment]
Ok it is quite rightly erroring since I changed the default for Ingress path from "" to "/". I'll change that in the schema and update the UPDATING.rst file. The other error seems to be about an unrelated SQLAlchemy deprecation though so I'll ignore that rather than make an unrelated change in this PR.

**Comment by cablespaghetti** [comment]
I don't understand the test failure we've currently got. 😞

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
> I don't understand the test failure we've currently got. 😞

Most likely a flaky failure, let's try it again.

**Comment by jedcunningham** [comment]
Thanks @cablespaghetti!

**Pull request merged**